src/librustdoc/html/static/js/search.js

```diff
@@ -167,33 +484,141 @@ window.initSearch = function(rawSearchIndex) {
     }
 
     /**
-     * Executes the query and returns a list of results for each results tab.
-     * @param  {Object}        query          - The user query
-     * @param  {Array<string>} searchWords    - The list of search words to query against
-     * @param  {string}        [filterCrates] - Crate to search in
-     * @return {{
-     *   in_args: Array<?>,
-     *   returned: Array<?>,
-     *   others: Array<?>
-     * }}
+     * Parses the query.
+     *
+     * The supported syntax by this parser is as follow:
+     *
+     * ident = *(ALPHA / DIGIT / "_")
+     * path = ident *(DOUBLE-COLON ident)
+     * arg = path [generics]
+     * arg-without-generic = path
+     * nonempty-arg-list = arg *WS *(COMMA/WS *WS arg)
+     * nonempty-arg-list-without-generics = arg-without-generic *WS *(COMMA/WS *WS arg
+     *                                                                without-generic)
+     * generics = OPEN-ANGLE-BRACKET *WS [ nonempty-arg-list-without-generics ] *WS
+     *            CLOSE-ANGLE-BRACKET
+     * return-args = RETURN-ARROW *WS nonempty-arg-list
+     *
+     * exact-search = [type-filter *WS COLON] *WS QUOTE ident QUOTE *WS [generics]
+     * type-search = [type-filter *WS COLON] *WS path *WS generics
```

Comment:
Something I notice by reading the parser output is how it can return a "mutli-element" query. I can't find any way to represent it in this grammar, but here's a suggestion that seems to match.

```suggestion
     * type-search = [type-filter *WS COLON] *WS nonempty-arg-list
```

Reply:
Good catch!